apiAuditLogs/apiAuditLogs.go

```diff
@@ -0,0 +1,329 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package apiAuditLogs
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "apiAuditLogs",
+	Doc:  "check the audit logs usage in the API",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	whiteList := map[string]bool{
+		"autocompleteChannelsForTeam":          true,
```

Comment:
@wiggin77 please, take a look to this list of whitelisted operations, that are the API handlers that have not Audit logs for now, probably some of them should be audited.

Reply:
Thanks @jespino for generating that list.  On review, I can see these should have auditing added:  

`disableBot, enableBot, executeCommand, generateMfaSecret, getAudits, getComplianceReport, getComplianceReports, getLogs, getPublicFile, getUserAudits, logout, postLog`

I'll add them in a new PR.

Reply:
I goint to remove them from the whiteList right away, and when everything is fixed we can add the check to the mattermost-server Makefile.

By the way, this is only checking for the api, not for the app layer.

Reply:
Done